bee-common/src/shutdown/mod.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+use crate::worker::Error as WorkerError;
+
+use async_std::task;
+use futures::channel::oneshot;
+use thiserror::Error;
+
+use std::result::Result;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Sending the shutdown signal to a worker failed.")]
+    SendingShutdownSignalFailed,
+
+    #[error("Waiting for worker to shut down failed.")]
+    WaitingforWorkerShutdownFailed(#[from] WorkerError),
+}
+
+pub type ShutdownNotifier = oneshot::Sender<()>;
+pub type ShutdownListener = oneshot::Receiver<()>;
```

Comment:
Out of curiosity, why are we not sticking to the usual names ? Like `ShutdownSender` and `ShutdownReceiver` ?

Reply:
good question: the reason is that semantically `ShutdownSender` and `ShutdownReceiver` make no sense, but `ShutdownNotifier` and `ShutdownListener` do as you can *notify* about a shutdown and you can *listen* for such an event, but you cannot literally send or receive a shutdown.

Reply:
I disagree but I also don't care, was just curious :)

Reply:
:joy: